src/solver/variable/economy/bindingConstraintsMarginalCost.h

```diff
@@ -323,12 +323,42 @@ class BindingConstMarginCost
         }
     }
 
+    void buildSurveyReport(SurveyResults& results,
+                           int dataLevel,
+                           int fileLevel,
+                           int precision) const
+    {
+        // Building syntheses results
+        // ------------------------------
+
+        // And only if we match the current data level _and_ precision level
+        if ((dataLevel & VCardType::categoryDataLevel) && (fileLevel & VCardType::categoryFileLevel)
+            && (precision & VCardType::precision))
+        {
+            results.isPrinted = isPrinted;
+            results.isCurrentVarNA[0] = isCurrentOutputNonApplicable(precision);
+            results.variableCaption = getBindConstraintCaption();
+
+            VariableAccessorType::template BuildSurveyReport<VCardType>(
+              results, pResults, dataLevel, fileLevel, precision);
+        }
+    }
+
 private:
+    // Private methods
+    // ---------------
     std::string getBindConstraintCaption() const
     {
         return associatedBC_->name + " (" + associatedBC_->operatorType + ")";
     }
 
+    bool isCurrentOutputNonApplicable(int precision) const
+    {
+        return precision < pow(2, associatedBC_->type - 1);
+    }
+
+    // Private data mambers
```

Comment:
Use dedicated keyword ?

Reply:
What do you mean ?

Reply:
```suggestion
    private:
```

Reply:
the private keyword is already there, but there is one **private** for data and methods